app/components/scbd-angularjs-services/filters/scbd-filters.js

```diff
@@ -30,6 +30,16 @@ import '../services/locale';
     };		
   }]);
 
+  //============================================================
+  //
+  //
+  //
+  //============================================================
+  app.filter('escape', function() {
+    return function(value) {
+      return value.replace(/[!"#$%&'()*+,.\/:;<=>?@[\\\]^`{|}~]/g, '\\$&');
+    };
+  });
```

Comment:
escape is very generic name. cna we use more mesaningful?

eg: `escapeHtmlAttribuetId`

I would suggest to reverse the regex with only allowed char and replace all other ones with underscore using 

```jsvascript 
/[^-_:.a-z0-9]/ig
```
The fisrt `^` insinde `[ ]` mean NOT. therefore the above regex mean: matche everything except follwing allowed chars: `-` `_` `:` `.` `a` to `z` and `0` to `9`

base on W3C: ID and Name attribute value valid char are: 
> ID and NAME tokens must begin with a letter ([A-Za-z]) and may be followed by any number of letters, digits ([0-9]), hyphens ("-"), underscores ("_"), colons (":"), and periods (".").
https://www.w3.org/TR/html4/types.html#:~:text=ID%20and%20NAME%20tokens%20must,tokens%20defined%20by%20other%20attributes.

therefore I would suggest follwing code:
```suggestion
  app.filter('escapeHtmlAttribuetId', function() {
    return function(value) {
      return value.replace(/[^-_:.a-z0-9]/gi, '_');
    };
  });
```

Reply:
done as suggested